### PR TITLE
Add tutorial about setting external docker registry in serverless

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:
- Add tutorial about setting external docker registry in serverless.
- Code and changed resources are in  https://github.com/kyma-project/kyma/pull/8888

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/8204

### CODE REVIEW

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+    ```
+
+6. Create the service principal with rights scoped to the Azure CR. Run:
+
+    ```bash
+    export SP_PASSWORD=$(az ad sp create-for-rbac --name http://${AZ_SERVICE_PRINCIPAL_NAME} --scopes ${AZ_REGISTRY_ID} --role ${ROLE} --query password --output tsv)
+    export SP_APP_ID=$(az ad sp show --id http://${AZ_SERVICE_PRINCIPAL_NAME} --query appId --output tsv)
+    ```
+
+   Or assign the desired role to the existing service principal. Run:
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:186]:
```suggestion
   Alternatively, assign the desired role to the existing service principal. Run:
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  dockerRegistry.password: $(echo "${SP_PASSWORD}" | base64)
+  dockerRegistry.serverAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+  dockerRegistry.registryAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+EOF
+```
+
+  </details>
+</div>
+
+> **CAUTION:** If you want to set external Docker Registry before you install Kyma, you need to manually add the Secret to the `installer-config-local.yaml.tpl` (local installation) or `installer-config-production.yaml.tpl` (production installation) template located under the `installation/resources` subfolder before you run the installation script.
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:282]:
```suggestion
> **CAUTION:** If you want to set an external Docker registry before you install Kyma, you need to manually add the Secret to the `installer-config-local.yaml.tpl` (local installation) or `installer-config-production.yaml.tpl` (production installation) template located under the `installation/resources` subfolder before you run the installation script.
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+### Trigger installation
+
+Trigger Kyma installation or update by labeling the Installation custom resource. Run:
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:286]:
```suggestion
Trigger Kyma installation or update it by labeling the Installation custom resource. Run:
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+<div tabs name="create-required-cloud-resources" group="external-docker-registry">
+  <details>
+  <summary label="docker-hub">
+  Docker Hub
+  </summary>
+
+1. Run the `export {VARIABLE}={value}` command to set up the following environment variables, where:
+
+    - **USERNAME** is the name of account in Docker Hub.
+    - **PASSWORD** is the password for account in Docker Hub.
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:58]:
```suggestion
    - **PASSWORD** is the password for the account in the Docker Hub.
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+    - **SERVER_ADDRESS** is the server address of Docker Hub. At the moment Kyma supports only `https://index.docker.io/v1/` server address.
+    - **REGISTRY_ADDRESS** is the registry address in Docker Hub (usually this is an account name).
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:60]:
```suggestion
    - **REGISTRY_ADDRESS** is the registry address in the Docker Hub.
    
    > **TIP:** Usually the Docker registry address is the same as the account name.
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+    export REGISTRY_ADDRESS=kyma-rocks
+    ```
+
+  </details>
+  <details>
+  <summary label="gcr">
+  GCR
+  </summary>
+
+To use Google Container Registry (GCR), create a Google service account that has a private key and the **Storage Admin** role permissions. Follow these steps:
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:77]:
```suggestion
To use GCR, create a Google service account that has a private key and the **Storage Admin** role permissions. Follow these steps:
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+    export GCS_KEY_JSON=$(< "$SECRET_FILE" base64 | tr -d '\n')
+    ```
+
+  </details>
+  <details>
+  <summary label="acr">
+  ACR
+  </summary>
+
+Create an Azure Container Registry (ACR) and a service principal. Follow these steps:
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:135]:
```suggestion
Create an ACR and a service principal. Follow these steps:
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+  dockerRegistry.password: $(echo "${SP_PASSWORD}" | base64)
+  dockerRegistry.serverAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+  dockerRegistry.registryAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+EOF
+```
+
+  </details>
+</div>
+
+> **CAUTION:** If you want to set an external Docker registry before you install Kyma, you need to manually add the Secret to the `installer-config-local.yaml.tpl` (local installation) or `installer-config-production.yaml.tpl` (production installation) template located under the `installation/resources` subfolder before you run the installation script.
```
Comment (kazydek) [inline docs/serverless/08-05-set-external-docker-registry.md:286]:
```suggestion
> **CAUTION:** If you want to set an external Docker registry before you install Kyma, manually add the Secret to the `installer-config-local.yaml.tpl` (local installation) or `installer-config-production.yaml.tpl` (production installation) template located under the `installation/resources` subfolder before you run the installation script.
```

File: docs/serverless/08-05-set-external-docker-registry.md [hunk 9]
```diff
@@ -0,0 +1,296 @@
...
+  dockerRegistry.password: $(echo "${SP_PASSWORD}" | base64)
+  dockerRegistry.serverAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+  dockerRegistry.registryAddress: $(echo "${AZ_REGISTRY_NAME}.${SERVER_ADDRESS}" | base64)
+EOF
+```
+
+  </details>
+</div>
+
+> **CAUTION:** If you want to set an external Docker registry before you install Kyma, manually add the Secret to the `installer-config-local.yaml.tpl` (local installation) or `installer-config-production.yaml.tpl` (production installation) template located under the `installation/resources` subfolder before you run the installation script.
```
Comment (michal-hudy) [inline docs/serverless/08-05-set-external-docker-registry.md:288]: No, you shouldn't modify `*.tpl` files, simply apply secret with an override to the cluster before triggering the installation.

**Pull request merged**